### PR TITLE
utils.auxiliary fixes

### PR DESCRIPTION
**Pull request by idanmiara**
## What does this PR do?

Fix some bugs in utils.auxiliary
Add some types hinting, rename PathOrDS

### CODE REVIEW

File: gdal/swig/python/osgeo/utils/auxiliary/color_palette.py [hunk 1]
```diff
@@ -352,13 +352,10 @@ def get_color_palette(color_palette_or_path_or_strings: ColorPaletteOrPathOrStri
     return pal
 
 
-def test_xml():
-    dir_path = 'sample/color_files'
+def test_xml(dir_path):
```
Comment (rcoup) [inline gdal/swig/python/osgeo/utils/auxiliary/color_palette.py:355]: Is this actually used? AFAICS the only caller was below

**Review by rcoup — APPROVED** [review]
LGTM! 😄  One question about an unused function.

File: gdal/swig/python/osgeo/utils/auxiliary/color_palette.py (hunk 1 above)
Comment (idanmiara) [reply gdal/swig/python/osgeo/utils/auxiliary/color_palette.py:355]: Thanks for your comments.
I was planning to add a test to use this function with the actual test files but I didn't get to it yet.
Meanwhile I wanted to replace the reference to this hardcoded dir which only existed in my local setup.

Comment (rcoup) [reply gdal/swig/python/osgeo/utils/auxiliary/color_palette.py:355]: Suggest to drop it out and re-add on the appropriate future PR.

Comment (idanmiara) [reply gdal/swig/python/osgeo/utils/auxiliary/color_palette.py]: > Suggest to drop it out and re-add on the appropriate future PR.

done. thanks!

Comment (rcoup) [reply gdal/swig/python/osgeo/utils/auxiliary/color_palette.py]: [flake8 is failing](https://github.com/OSGeo/gdal/pull/3404/checks?check_run_id=1862571265) on `import glob` being unused

Comment (idanmiara) [reply gdal/swig/python/osgeo/utils/auxiliary/color_palette.py]: Thanks

**Pull request merged**